### DISCUSSION TITLE
Why should we include environment variables in our repository

### DISCUSSION THREAD

**Discussion by IDrumsey** [vercel/next.js#45310] [Help] [answered]
The [docs](https://nextjs.org/docs/basic-features/environment-variables#:~:text=Note%3A%20.env%2C%20.env.development%2C%20and%20.env.production%20files%20should%20be%20included%20in%20your%20repository%20as%20they%20define%20defaults.%20.env*.local%20should%20be%20added%20to%20.gitignore%2C%20as%20those%20files%20are%20intended%20to%20be%20ignored.%20.env.local%20is%20where%20secrets%20can%20be%20stored.) mention tracking the environment variable files like `.env.production` and `.env.development`. They say these files contain "defaults" and I don't really know what is meant by that. Wouldn't these files have sensitive secrets in them that **shouldn't** be known by the repository? Are we just not supposed to have secrets in the non-`.env.local` files? Maybe my understanding is incorrect, any clarification is appreciated.

**Comment by kfirfitousi** [comment, marked as answer]
`.env.production` and `.env.development` can be used for non-sensitive defaults.  
Something like this:

```env
# .env.production
NEXT_PUBLIC_SITE_URL=https://www.acme.com
```

```env
# .env.development
NEXT_PUBLIC_SITE_URL=https://localhost:3000
```

Secrets should go in `.env.local` which is meant to be included in `.gitignore`

**Reply by IDrumsey** [reply]
Is there a way to have different environment variables that **are** secrets?
